[Lin Yuxiang] iP

Clin-lyx opened the pull request:
# DUKE
> “Your mind is for having ideas, not holding them.” – David Allen

DukePro frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- *SUPER* fast to learn

All you need to do is,

1. copy it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/349)
2. double click it
3. add your tasks.
4. let it manage your tasks for you. :+1:

Use `list` to list all tasks that have been added.
And it's **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders

 If you Java programmer, you can use it to practice Java too. Here's the main method for the launcher:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Clin-lyx committed:
Level 0. Rename, Greet, Exit

Clin-lyx committed:
Level 1. Echo

Clin-lyx committed:
Level 2. Add, List

Clin-lyx committed:
Level 3. Mark as Done

Clin-lyx committed:
Level 4. ToDos, Events, Deadlines

Clin-lyx committed:
Level 5. Handle Errors

Clin-lyx committed:
Level 6. Delete

Clin-lyx committed:
Set up necessary files for TextUITesting

Clin-lyx committed:
Add FileHandler.java

Update Duke.java Event.java Task.java, ToDo.java, Deadline.java to
accomodate FileHandler.java.

A DukeContentWrongFormat exception to check if the txt file has
unrecognizable tasks.

Clin-lyx committed:
Merge branch 'branch-Level-7'

Clin-lyx committed:
Implement TimeProcessir.java

The TimeProcessor.java handle the task to convert a string to a sepcific
string. However, string that are abbreviation of days of week or
month cannot be converted.

Clin-lyx committed:
Merge branch 'branch-Level-8'

Clin-lyx committed:
Refactor functionalities of Duke.java

Creating Parser.java, TaskList.java, and Ui.java toseparate the
functionalities of Duke.java

Clin-lyx committed:
Merge branch 'More-OOP'

Clin-lyx committed:
Divide classes into packages

Clin-lyx committed:
Merge branch 'packages'

Clin-lyx committed:
Merge branch 'add-gradle-support'

Clin-lyx committed:
Set up Gradle to IP

The gradle can now run and build the project

kiwibang commented on src/main/java/duke/processors/TimeProcessor.java:
```diff
@@ -0,0 +1,55 @@
+           } catch (IllegalArgumentException e) {
+               if (info.isEmpty()){
+                   date = current;
+               } else {
```
Perhaps this can be converted to `else if (!Character.isDigit(info.charAt(0)))`? 🤔 
It would reduce 1 layer of nesting!

kiwibang commented on src/main/java/duke/processors/Ui.java:
```diff
@@ -0,0 +1,52 @@
+import duke.exception.DukeEmptyCommandException;
+import java.util.Scanner;
+public class Ui {
+    private String separation = "_______________________________________________";
```
Perhaps this could be a final String? 🤔 
Maybe could also capitalise the constant name? 

For example:
```java
private final String SEPARATION = "_______________________________________________";
```

kiwibang commented on src/main/java/duke/processors/Ui.java:
```diff
@@ -0,0 +1,52 @@
+        System.out.println(separation);
+    }
+
+    public void OnExit() {
```
Shoule this method name be in camelCase?
For example:
```java
public void onExit()
```

kiwibang commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,26 @@
+import duke.exception.DukeNoDescriptionException;
+
+public class ToDo extends Task{
+    public ToDo(String Description) throws DukeNoDescriptionException{
```
Perhaps the curly brace could be space separated? 
For example:
```java
public ToDo(String Description) throws DukeNoDescriptionException {
```

kiwibang commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+        this.isDone = false;
+    }
+
+    private String GetStatusIcon() {
```
Very clean way of getting the status! 👍

kiwibang commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,39 @@
+package duke.task;
+
+import duke.processors.FileHandler;
```
perhaps this should be separated? 
For example:
```java
import duke.processors.FileHandler;

public class Task {
```

kiwibang reviewed (commented):
LGTM! Just some nits to fix. Better than my own PR honestly 👍

Clin-lyx committed:
Add Junit test

Clin-lyx committed:
Merge branch 'junit'

Clin-lyx committed:
Create duke.jar

Clin-lyx committed:
Merge branch 'jar'

Clin-lyx committed:
Add javadoc to the java files

Clin-lyx committed:
Set files to coding standard

Clin-lyx committed:
Add feature find

Clin-lyx committed:
Merge branch 'branch-Level-9'

conradsoon commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,41 @@
+        PARSER = new Parser(tasks);
+    }
+
+    private void run() {
```
Can consider adding some JavaDocs comments?

conradsoon commented on src/main/java/duke/processors/Parser.java:
```diff
@@ -0,0 +1,107 @@
+    private final TaskList TASKS;
+    private final FileHandler DUKE = new FileHandler();
+    private final Ui UI = new Ui();
+    private boolean terminate;
```
Consider renaming the variable "terminate" to "isTerminated"?

conradsoon commented on src/main/java/duke/processors/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+ */
+public class TaskList {
+
+    private final ArrayList<Task> TASKS = new ArrayList<>();
```
Consider making the variable name "TASKS" use camelCase.

conradsoon commented on src/main/java/duke/processors/TaskList.java:
```diff
@@ -0,0 +1,90 @@
+     * Add a task to the ArrayList.
+     * @param task the task to be added
+     */
+    public void addTasks(Task task) {
```
Consider making the method name "addTasks" singular since it adds only one task at a time?

conradsoon reviewed (commented):
Overall, the code follows most of the provided coding standards. Great job!

Clin-lyx committed:
Build GUI

Clin-lyx committed:
Add Assertion

Added assertions into mainly processor and task package to ensure
no bad input or output provided to the function.

Clin-lyx committed:
Merge pull request #2 from Clin-lyx/branch-A-Assertions

Add Assertion to Java packages.

Clin-lyx committed:
Edit code into better quality

Adjustments on javadoc to better suit the coding standard.

Clin-lyx committed:
Merge pull request #3 from Clin-lyx/branch-A-codeQuality

Edit code into better quality

Clin-lyx committed:
Add bcd extension

Added C-Natural dates to the code, allowing input like mon tue etc.
Also 24hr time will be converted to 12hr time.

Clin-lyx committed:
Merge branch 'branch-bcd-extension'

Clin-lyx committed:
Upload Ui.png to docs file

Added several improvement to tasks and Ui

Clin-lyx committed:
Improvement on FileHandler

Clin-lyx committed:
Add title to the main window

Modifications made in build.gradle

Clin-lyx committed:
Update README.md

Clin-lyx committed:
Update README.md

Clin-lyx committed:
Correct chat box name

Clin-lyx committed:
Update README.md

Clin-lyx committed:
Update README.md
